### PR TITLE
net: move IPv4/6BUF define to common header 

### PR DESCRIPTION
**Pull request by anchao**
## Summary

net: move IPv4/6BUF define to common header

Signed-off-by: chao an <anchao@xiaomi.com>

## Impact

N/A

## Testing

tcp/udp iperf test

### CODE REVIEW

File: arch/arm/src/tiva/lm/lm3s_ethernet.c [hunk 1]
```diff
@@ -675,10 +671,11 @@ static int tiva_txpoll(struct net_driver_s *dev)
 
 static void tiva_receive(struct tiva_driver_s *priv)
 {
+  struct net_driver_s *dev = &priv->ld_dev;
+  int      bytesleft;
+  int      pktlen;
```
Comment (pkarashchenko) [inline arch/arm/src/tiva/lm/lm3s_ethernet.c:676]:
```suggestion
  int bytesleft;
  int pktlen;
```

File: drivers/net/slip.c [hunk 2]
```diff
@@ -111,8 +111,8 @@
 
 /* This is a helper pointer for accessing the contents of the ip header */
 
-#define IPv4BUF        ((FAR struct ipv4_hdr_s *)priv->dev.d_buf)
-#define IPv6BUF        ((FAR struct ipv6_hdr_s *)priv->dev.d_buf)
+#define SLIPIPv4BUF        ((FAR struct ipv4_hdr_s *)priv->dev.d_buf)
+#define SLIPIPv6BUF        ((FAR struct ipv6_hdr_s *)priv->dev.d_buf)
```
Comment (pkarashchenko) [inline drivers/net/slip.c:115]:
```suggestion
#define SLIPIPv4BUF    ((FAR struct ipv4_hdr_s *)priv->dev.d_buf)
#define SLIPIPv6BUF    ((FAR struct ipv6_hdr_s *)priv->dev.d_buf)
```

File: net/icmpv6/icmpv6.h [hunk 3]
```diff
@@ -52,6 +52,10 @@
 #define icmpv6_callback_free(dev, conn, cb) \
   devif_conn_callback_free((dev), (cb), &(conn)->sconn.list, &(conn)->sconn.list_tail)
 
+/* This is a helper pointer for accessing the contents of the IP header */
+
+#define ICMPv6BUF(hl) ((FAR void *)&dev->d_buf[NET_LL_HDRLEN(dev) + hl])
```
Comment (pkarashchenko) [inline net/icmpv6/icmpv6.h:57]:
```suggestion
#define ICMPv6BUF(hl) ((FAR void *)&dev->d_buf[NET_LL_HDRLEN(dev) + (hl)])
```

File: net/igmp/igmp_input.c [hunk 4]
```diff
@@ -115,7 +108,7 @@
 
 void igmp_input(struct net_driver_s *dev)
 {
-  FAR struct igmp_iphdr_s *ipv4 = IPv4BUF;
+  FAR struct igmp_iphdr_s *ipv4 = (FAR struct igmp_iphdr_s *)IPv4BUF;
```
Comment (pkarashchenko) [inline net/igmp/igmp_input.c:111]:
```suggestion
  FAR struct igmp_iphdr_s *ipv4 = IGMPBUF(0);
```

File: net/igmp/igmp_send.c [hunk 5]
```diff
@@ -104,7 +102,7 @@ static uint16_t igmp_chksum(FAR uint8_t *buffer, int buflen)
 void igmp_send(FAR struct net_driver_s *dev, FAR struct igmp_group_s *group,
                FAR const in_addr_t *destipaddr, uint8_t msgid)
 {
-  FAR struct igmp_iphdr_s *ipv4 = IPv4BUF;
+  FAR struct igmp_iphdr_s *ipv4 = (FAR struct igmp_iphdr_s *)IPv4BUF;
```
Comment (pkarashchenko) [inline net/igmp/igmp_send.c:105]:
```suggestion
  FAR struct igmp_iphdr_s *ipv4 = IGMPBUF(0);
```

Comment (anchao) [reply net/igmp/igmp_send.c:105]: Done

File: net/igmp/igmp_input.c (hunk 4 above)
Comment (anchao) [reply net/igmp/igmp_input.c]: Done

File: drivers/net/slip.c (hunk 2 above)
Comment (anchao) [reply drivers/net/slip.c]: Done

File: arch/arm/src/lpc54xx/lpc54_ethernet.c [hunk 6]
```diff
@@ -251,7 +251,6 @@
  * header.
  */
 
-#define ETHBUF       ((struct eth_hdr_s *)priv->eth_dev.d_buf)
 #define ETH8021QWBUF ((struct eth_8021qhdr_s *)priv->eth_dev.d_buf)
```
Comment (xiaoxiang781216) [inline arch/arm/src/lpc54xx/lpc54_ethernet.c:254]: should we move this to the common place too

File: drivers/net/loopback.c [hunk 7]
```diff
@@ -61,8 +61,8 @@
 
 /* This is a helper pointer for accessing the contents of the IP header */
 
-#define IPv4BUF ((FAR struct ipv4_hdr_s *)priv->lo_dev.d_buf)
-#define IPv6BUF ((FAR struct ipv6_hdr_s *)priv->lo_dev.d_buf)
+#define LOIPv4BUF ((FAR struct ipv4_hdr_s *)priv->lo_dev.d_buf)
```
Comment (xiaoxiang781216) [inline drivers/net/loopback.c:64]: why not reuse IPv4BUF and IPv6BUF?

File: drivers/net/slip.c [hunk 8]
```diff
@@ -111,8 +111,8 @@
 
 /* This is a helper pointer for accessing the contents of the ip header */
 
-#define IPv4BUF        ((FAR struct ipv4_hdr_s *)priv->dev.d_buf)
-#define IPv6BUF        ((FAR struct ipv6_hdr_s *)priv->dev.d_buf)
+#define SLIPIPv4BUF    ((FAR struct ipv4_hdr_s *)priv->dev.d_buf)
```
Comment (xiaoxiang781216) [inline drivers/net/slip.c:114]: ditto

Comment (anchao) [reply drivers/net/slip.c]: Done

File: drivers/net/loopback.c (hunk 7 above)
Comment (anchao) [reply drivers/net/loopback.c]: Done

File: arch/arm/src/lpc54xx/lpc54_ethernet.c (hunk 6 above)
Comment (anchao) [reply arch/arm/src/lpc54xx/lpc54_ethernet.c:254]: Done

File: net/tcp/tcp_input.c [hunk 9]
```diff
@@ -933,22 +923,22 @@ static void tcp_input(FAR struct net_driver_s *dev, uint8_t domain,
...
                       {
                         uint16_t tcp_mss = TCP_MSS(dev, iplen);
 
                         /* An MSS option with the right option length. */
 
                         tmp16 =
-                          (dev->d_buf[hdrlen + 2 + i] << 8) |
-                          dev->d_buf[hdrlen + 3 + i];
+                          (IPDATA(tcpiplen + 2 + i) << 8) |
+                          IPDATA(tcpiplen + 3 + i);
```
Comment (pkarashchenko) [inline net/tcp/tcp_input.c:934]: can be combined to two lines instead of 3

**Review by pkarashchenko — APPROVED** [review]

File: net/mld/mld_send.c [hunk 10]
```diff
@@ -61,17 +61,16 @@
 
 /* Buffer layout */
 
-#define IPv6BUF     ((FAR struct ipv6_hdr_s *)&dev->d_buf[NET_LL_HDRLEN(dev)])
 #define RABUF       ((FAR struct ipv6_router_alert_s *) \
-                     (&dev->d_buf[NET_LL_HDRLEN(dev)] + IPv6_HDRLEN))
+                    IPBUF(IPv6_HDRLEN)
```
Comment (pkarashchenko) [inline net/mld/mld_send.c:65]:
```suggestion
                    IPBUF(IPv6_HDRLEN))
```

File: net/mld/mld_send.c [hunk 11]
```diff
@@ -61,17 +61,16 @@
...
 #define QUERYBUF    ((FAR struct mld_mcast_listen_query_s *) \
-                     (&dev->d_buf[NET_LL_HDRLEN(dev)] + MLD_HDRLEN))
+                    IPBUF(MLD_HDRLEN)
```
Comment (pkarashchenko) [inline net/mld/mld_send.c:67]:
```suggestion
                    IPBUF(MLD_HDRLEN))
```

File: net/mld/mld_send.c [hunk 12]
```diff
@@ -61,17 +61,16 @@
...
 #define V1REPORTBUF ((FAR struct mld_mcast_listen_report_v1_s *) \
-                     (&dev->d_buf[NET_LL_HDRLEN(dev)] + MLD_HDRLEN))
+                    IPBUF(MLD_HDRLEN)
```
Comment (pkarashchenko) [inline net/mld/mld_send.c:69]:
```suggestion
                    IPBUF(MLD_HDRLEN))
```

File: net/mld/mld_send.c [hunk 13]
```diff
@@ -61,17 +61,16 @@
...
 #define V2REPORTBUF ((FAR struct mld_mcast_listen_report_v2_s *) \
-                     (&dev->d_buf[NET_LL_HDRLEN(dev)] + MLD_HDRLEN))
+                    IPBUF(MLD_HDRLEN)
```
Comment (pkarashchenko) [inline net/mld/mld_send.c:71]:
```suggestion
                    IPBUF(MLD_HDRLEN))
```

File: net/mld/mld_send.c [hunk 14]
```diff
@@ -61,17 +61,16 @@
...
 #define DONEBUF     ((FAR struct mld_mcast_listen_done_s *) \
-                     (&dev->d_buf[NET_LL_HDRLEN(dev)] + MLD_HDRLEN))
+                    IPBUF(MLD_HDRLEN)
```
Comment (pkarashchenko) [inline net/mld/mld_send.c:73]:
```suggestion
                    IPBUF(MLD_HDRLEN))
```

File: net/tcp/tcp_input.c (hunk 9 above)
Comment (anchao) [reply net/tcp/tcp_input.c:934]: Done!

File: net/mld/mld_send.c (hunk 10 above)
Comment (anchao) [reply net/mld/mld_send.c:65]: Done!

File: net/mld/mld_send.c (hunk 12 above)
Comment (anchao) [reply net/mld/mld_send.c:69]: Done!

File: net/mld/mld_send.c (hunk 11 above)
Comment (anchao) [reply net/mld/mld_send.c:67]: Done!

File: net/mld/mld_send.c (hunk 13 above)
Comment (anchao) [reply net/mld/mld_send.c:71]: Done!

File: net/mld/mld_send.c (hunk 14 above)
Comment (anchao) [reply net/mld/mld_send.c:73]: Done!

File: net/mld/mld_send.c [hunk 15]
```diff
@@ -61,17 +61,16 @@
 
 /* Buffer layout */
 
-#define IPv6BUF     ((FAR struct ipv6_hdr_s *)&dev->d_buf[NET_LL_HDRLEN(dev)])
 #define RABUF       ((FAR struct ipv6_router_alert_s *) \
-                     (&dev->d_buf[NET_LL_HDRLEN(dev)] + IPv6_HDRLEN))
+                    IPBUF(IPv6_HDRLEN))
```
Comment (xiaoxiang781216) [inline net/mld/mld_send.c:65]: merge

File: net/mld/mld_send.c [hunk 16]
```diff
@@ -61,17 +61,16 @@
...
 #define QUERYBUF    ((FAR struct mld_mcast_listen_query_s *) \
-                     (&dev->d_buf[NET_LL_HDRLEN(dev)] + MLD_HDRLEN))
+                    IPBUF(MLD_HDRLEN))
```
Comment (xiaoxiang781216) [inline net/mld/mld_send.c:67]: merge

File: net/mld/mld_send.c [hunk 17]
```diff
@@ -61,17 +61,16 @@
...
 #define V1REPORTBUF ((FAR struct mld_mcast_listen_report_v1_s *) \
-                     (&dev->d_buf[NET_LL_HDRLEN(dev)] + MLD_HDRLEN))
+                    IPBUF(MLD_HDRLEN))
```
Comment (xiaoxiang781216) [inline net/mld/mld_send.c:69]: merge

File: net/mld/mld_send.c [hunk 18]
```diff
@@ -61,17 +61,16 @@
...
 #define V2REPORTBUF ((FAR struct mld_mcast_listen_report_v2_s *) \
-                     (&dev->d_buf[NET_LL_HDRLEN(dev)] + MLD_HDRLEN))
+                    IPBUF(MLD_HDRLEN))
```
Comment (xiaoxiang781216) [inline net/mld/mld_send.c:71]: merge

File: net/mld/mld_send.c [hunk 19]
```diff
@@ -61,17 +61,16 @@
...
 #define DONEBUF     ((FAR struct mld_mcast_listen_done_s *) \
-                     (&dev->d_buf[NET_LL_HDRLEN(dev)] + MLD_HDRLEN))
+                    IPBUF(MLD_HDRLEN))
```
Comment (xiaoxiang781216) [inline net/mld/mld_send.c:73]: merge

File: net/mld/mld_send.c (hunk 16 above)
Comment (anchao) [reply net/mld/mld_send.c:67]: Done

File: net/mld/mld_send.c (hunk 17 above)
Comment (anchao) [reply net/mld/mld_send.c:69]: Done

File: net/mld/mld_send.c (hunk 19 above)
Comment (anchao) [reply net/mld/mld_send.c]: Done

File: net/sixlowpan/sixlowpan_input.c [hunk 20]
```diff
@@ -616,11 +615,14 @@ static int sixlowpan_frame_process(FAR struct radio_driver_s *radio,
 
 static int sixlowpan_dispatch(FAR struct radio_driver_s *radio)
 {
+#ifdef CONFIG_NET_6LOWPAN_DUMPBUFFER
+  struct net_driver_s *dev = &radio->r_dev;
```
Comment (pkarashchenko) [inline net/sixlowpan/sixlowpan_input.c:619]: Please put this below `int ret;` and hide `sixlowpan_dumpbuffer` under `#ifdef CONFIG_NET_6LOWPAN_DUMPBUFFER`

**Review by pkarashchenko — COMMENTED** [review]
Please fix style issues

File: net/sixlowpan/sixlowpan_input.c (hunk 20 above)
Comment (anchao) [reply net/sixlowpan/sixlowpan_input.c:619]: Done

**Comment by anchao** [comment]
> Please fix style issues

Thank you, all addressed.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: net/mld/mld_send.c (hunk 15 above)
Comment (anchao) [reply net/mld/mld_send.c]: Done

File: arch/arm/src/lpc54xx/lpc54_ethernet.c [hunk 21]
```diff
@@ -934,6 +933,8 @@ static void lpc54_eth_reply(struct lpc54_ethdriver_s *priv)
 
 static void lpc54_eth_rxdispatch(struct lpc54_ethdriver_s *priv)
 {
+  struct net_driver_s *dev = &priv->eth_dev;
```
Comment (hartmannathan) [inline arch/arm/src/lpc54xx/lpc54_ethernet.c:936]: This variable (`dev`) is not used in this function. Perhaps you intended to change all `&priv->eth_dev` below?

File: arch/arm/src/tiva/lm/lm3s_ethernet.c [hunk 22]
```diff
@@ -675,10 +671,11 @@ static int tiva_txpoll(struct net_driver_s *dev)
 
 static void tiva_receive(struct tiva_driver_s *priv)
 {
+  struct net_driver_s *dev = &priv->ld_dev;
```
Comment (hartmannathan) [inline arch/arm/src/tiva/lm/lm3s_ethernet.c:674]: Ditto: Did you intend to replace all `priv->ld_dev.` below with `dev->`?

File: drivers/net/slip.c [hunk 23]
```diff
@@ -618,6 +614,7 @@ static int slip_rxtask(int argc, FAR char *argv[])
 
   /* Loop forever */
 
+  dev = &priv->dev;
```
Comment (hartmannathan) [inline drivers/net/slip.c:617]: Ditto: replace all `priv->dev.` with `dev->` below...

File: drivers/net/tun.c [hunk 24]
```diff
@@ -619,6 +614,8 @@ static void tun_net_receive_tap(FAR struct tun_device_s *priv)
 
 static void tun_net_receive_tun(FAR struct tun_device_s *priv)
 {
+  FAR struct net_driver_s *dev = &priv->dev;
```
Comment (hartmannathan) [inline drivers/net/tun.c:617]: Ditto

File: arch/arm/src/tiva/lm/lm3s_ethernet.c (hunk 22 above)
Comment (anchao) [reply arch/arm/src/tiva/lm/lm3s_ethernet.c:674]: Done

File: drivers/net/slip.c (hunk 23 above)
Comment (anchao) [reply drivers/net/slip.c:617]: Done

File: drivers/net/tun.c (hunk 24 above)
Comment (anchao) [reply drivers/net/tun.c:617]: Done

**Comment by pkarashchenko** [comment]
@hartmannathan please approve if you do not have other comments, so we can move forward with merge

**Comment by hartmannathan** [comment]
> @hartmannathan please approve if you do not have other comments, so we can move forward with merge

Reviewing now...

**Review by hartmannathan — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**